api/core/v1beta2/machinehealthcheck_types.go

```diff
@@ -110,14 +88,57 @@ type MachineHealthCheckSpec struct {
 	// +kubebuilder:validation:Minimum=0
 	NodeStartupTimeoutSeconds *int32 `json:"nodeStartupTimeoutSeconds,omitempty"`
 
-	// remediationTemplate is a reference to a remediation template
+	// unhealthyNodeConditions contains a list of conditions that determine
+	// whether a node is considered unhealthy. The conditions are combined in a
+	// logical OR, i.e. if any of the conditions is met, the node is unhealthy.
+	//
+	// +optional
+	// +listType=atomic
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=100
+	UnhealthyNodeConditions []UnhealthyNodeCondition `json:"unhealthyNodeConditions,omitempty"`
+}
+
+// MachineHealthCheckRemediation configures if and how remediations are triggered if a Machine is unhealthy.
+// +kubebuilder:validation:MinProperties=1
+type MachineHealthCheckRemediation struct {
+	// triggerIf configures if remediations are triggered.
+	// If this field is not set, remediations are always triggered.
+	// +optional
+	TriggerIf MachineHealthCheckRemediationTriggerIf `json:"triggerIf,omitempty,omitzero"`
```

Comment:
as an optional struct, should this be a pointer?

Reply:
> From an API best practices perspective pointer + omitempty or non-pointer + omitempty,omitzero works

This is quite a new thing, with Go 1.24 we can now use `omitzero`. So, as long as there is no valid use case for a `{}` meaning something different than omitted, the correct thing now is to be not a pointer, and have `omitzero`. This should make coding simpler for folks over time as we will have far fewer dereference points for structs

Reply:
ack. Does this then set a hard dep on clients needing to be on go 1.24?

Reply:
Yes. But we already have that because we rely on controller-runtime and k8.io verisons that require Go 1.24